libraries/AP_Torqeedo/AP_Torqeedo.cpp

```diff
@@ -90,25 +106,41 @@ bool AP_Torqeedo::init_internals()
     _uart->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
     _uart->set_unbuffered_writes(true);
 
-    // initialise onoff pin and set for 0.5 sec to turn on motor
-    if (_pin_onoff > -1) {
-        hal.gpio->pinMode(_pin_onoff, HAL_GPIO_OUTPUT);
-        hal.gpio->write(_pin_onoff, 1);
-
-        // delay 0.5 sec and then unset pin
-        hal.scheduler->delay(500);
-        hal.gpio->write(_pin_onoff, 0);
+    // if using tiller connection set on/off pin for 0.5 sec to turn on battery
+    if (_type == ConnectionType::TYPE_TILLER) {
+        if (_pin_onoff > -1) {
+            hal.gpio->pinMode(_pin_onoff, HAL_GPIO_OUTPUT);
+            hal.gpio->write(_pin_onoff, 1);
+            hal.scheduler->delay(500);
+            hal.gpio->write(_pin_onoff, 0);
+        } else {
```

Comment:
ok, added specific check that _pin == -1.

Reply:
Txs.  I guess this would catch the case where the user sets the pin to -2 or some other large negative number.  If I go down that route then I probably need to add a pre-arm check that they haven't set the param to an invalid value like this.. and I feel that's a bit too strict and/or not worth the flash cost so I might just leave it manipulating the CTS or RTS pins if the params are set to any negative value.